### PR TITLE
Repo Structure Spec based proposal

### PR DESCRIPTION
**Pull request by shemnon**
This aims to make a spec based proposal that also addresses the
feedbak given to prior proposals and to the specifics of the repolinter
that were not objected to by anyone else.

I also updated the repolinter to make it run against all projects.  Some
hackery with copying the repolinter into the local cloned repo was
required as repolinter does not allow pointing to an external config.

Required content is not automatically checked, but is part of the spec.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

### CODE REVIEW

File: repo_structure/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    --caliper:  Include Caliper repositories
+    --ursa:     Include Ursa repositories
+    --grid:     Include Grid repositories
+    --aries:    Include Aries repositories
+    --transact: Include Transact repositories
+    --avalon:   Include Avalon repositories
+    --projects: Include Project repositories
+    --labs:     Include Labs repositories
+    --other:    Include Other repositories
+    --gerrit:   Include Gerrit repositories
```
Comment (ryjones) [inline repo_structure/README.md]: not used any more

Comment (tkuhrt) [reply repo_structure/README.md]: Most likely common from [common/command-line.sh](https://github.com/hyperledger-labs/hyperledger-community-management-tools/blob/master/common/command-line.sh#L127). Looks like we have some cleanup to do in that file

File: repo_structure/repolint.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        "maintainers-file-exists:file-existence": ["error", {"files": ["MAINTAINERS.md"]}],
+        "contributing-file-exists:file-existence": ["error", {"files": ["CONTRIBUTING.md"]}],
+        "changelog-file-exists:file-existence": ["error", {"files": ["CHANGELOG.md"]}],
+        "test-directory-exists:directory-existence": ["error", {"directories": ["**/test*", "**/specs"], "nocase": true}],
+        "integrates-with-ci:file-existence": [
+            "error",
+            {
+              "files": [
+                "circle.yml", 
+                ".circleci/config.yml",
```
Comment (ryjones) [inline repo_structure/repolint.json]: "Jenkinsfile",
"Jenkinsfile.ci",
"Jenkinsfile.cd",

Comment (shemnon) [reply repo_structure/repolint.json]: Dnoe

File: repo_structure/README.md (hunk 1 above)
Comment (shemnon) [reply repo_structure/README.md]: That's exactly where it's coming from.

**Pull request merged**